homeassistant/components/fronius/__init__.py

```diff
@@ -204,10 +203,8 @@ async def _init_devices_inverter(self, _now: datetime | None = None) -> None:
             self.inverter_coordinators.append(_coordinator)
 
             # Only for re-scans. Initial setup adds entities through sensor.async_setup_entry
-            if self.sensor_async_add_entities is not None:
-                _coordinator.add_entities_for_seen_keys(
-                    self.sensor_async_add_entities, InverterSensor
-                )
+            if self.config_entry.state == ConfigEntryState.LOADED:
```

Comment:
I wonder if we should keep that distinction, or if it would be more readable to just instantiate all inverter coordinators via the dispatcher (on setup as well as newly found).

Reply:
My focus was on a small PR adressing the comment to separate the platform sensor. I'd suggest to keep it simple and go about refacting the mechanism in a 2nd step. We can use this to build on.

With the re-scheduling I actually thought of not having that check. I decided not to go down that path, because it would have blown up the PR complexity too much. 
Currently the platform is setup after `init_devices()` . The platform setup would need to happen first to have the listeners ready. I did a quick test with moving it around. Way too many tests failed because the entities do not appear during a time the tests expects them. We'd need to re-work the test, too. To ensure no side-effects occur, test it on a few live environments as well. This kind of refactoring I considered too complex to include as well.